otherlibs/systhreads/st_stubs.c

```diff
@@ -435,37 +430,23 @@ CAMLprim value caml_thread_cleanup(value unit)
 
 static void caml_thread_stop(void)
 {
-  caml_thread_t next;
-
   /* PR#5188, PR#7220: some of the global runtime state may have
      changed as the thread was running, so we save it in the
      curr_thread data to make sure that the cleanup logic
      below uses accurate information. */
   caml_thread_save_runtime_state();
-
-  next = Current_thread->next;
-
   /* The main domain thread does not go through [caml_thread_stop]. There is
      always one more thread in the chain at this point in time. */
-  CAMLassert(next != Current_thread);
-
-  caml_threadstatus_terminate(Terminated(Current_thread->descr));
-  caml_thread_remove_info(Current_thread);
-
-  /* FIXME: tricky bit with backup thread
-
-     Normally we expect the next thread to kick in and resume operation by
-     first setting Current_thread to the right TLS dec data. However it may
-     very well be that there's no runnable dec next (eg: next dec is
-     blocking.), so we set it to next for now to give a valid state to the
-     backup thread. */
-  Current_thread = next;
-
+  CAMLassert(this_thread->next != this_thread);
+  caml_threadstatus_terminate(Terminated(this_thread->descr));
+  /* This sets Current_thread to a sane value in case the backup
```

Comment:
I don't fully understand either why the code was needed earlier, and why it is OK to remove it now when `caml_thread_remove_info` remains the same.

Reply:
This was correct before; this issue is now more abstractly handled by thread_remove_info directly.